### PR TITLE
Remove discontinued and unmaintained boards

### PR DESCRIPTION
**Pull request by nashif**
Removing support for galileo, arduino_101 and Quark SE C1000 and their variants.

### CODE REVIEW

**Review by pfalcon — COMMENTED** [review]
arduino_101 is an important sample board for embedded x86 functionality, it's the default test board for some Zephyr-related projects, e.g. https://github.com/intel/zephyr.js .

I myself have a galileo lying somewhere around.

I believe there were commits for minnowboard until very recently (was it target of some hypervisor work or something?). 

Summing up, why remove well-known, popular (large amounts still in the community) and useful boards?

**Review by carlescufi — APPROVED** [review]
LGTM

**Comment by galak** [comment]
Shouldn't we be marking these boards as BOARD_DEPRECATED first?

File: tests/subsys/fs/fat_fs_api/README.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ Building and Running Project:
 The demo will run on Arduino 101 and will use the on-board SPI flash.
```
Comment (andrewboie) [inline tests/subsys/fs/fat_fs_api/README.txt]: still mentions Arduino 101

File: scripts/sanity_chk/sanity_compare.args [hunk 2]
```diff
@@ -8,11 +8,3 @@ frdm_k64f
...
 -p
 nrf52_pca10040
--p
-quark_d2000_crb
--p
-quark_se_c1000_devboard
--p
-quark_se_c1000_ss_devboard
--p
-altera_max10
```
Comment (andrewboie) [inline scripts/sanity_chk/sanity_compare.args]: why was altera_max10 removed? It's a Nios II board

**Comment by gnuless** [comment]
> arduino_101 is an important sample board for embedded x86 functionality,

It's been discontinued for two years.

> I believe there were commits for minnowboard until very recently (was it target of some hypervisor work or something?).

I'm not sure why the Minnowboard is being yanked. @nashif Minnowboard isn't a Quark, do we really want to drop support?

**Comment by gnuless** [comment]
> I believe there were commits for minnowboard until very recently (was it target of some hypervisor work or something?).

Sorry, @nashif, I think @pfalcon is mistaken and I didn't look closely, the Minnowboard isn't being removed here.

**Review by ghost — APPROVED** [review]

File: scripts/sanity_chk/sanity_compare.args (hunk 2 above)
Comment (nashif) [reply scripts/sanity_chk/sanity_compare.args]: altera_max10 is not being removed, here just cleaning up this more than I should have I guess.

**Review by tbursztyka — APPROVED** [review]
I am gladly approving that PR!

Goodbye awful galileo board...


oh and quark_se ones too, yes no more qmsi

**Review by jukkar — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
There's material out there about running Zephyr on some of these boards.  But I suppose we can aim them at the LTS (1.14) release instead of master.

**Review by wentongwu — APPROVED** [review]

**Pull request merged**